src/Icicle/Sea/FromAvalanche/Program.hs

```diff
@@ -205,7 +205,7 @@ seaOfXValue v t
       | ArrayT t' <- t
       -> let len = length vs
              writes arr (v',i)
-              = prim (PrimUpdate $ PrimUpdateArrayPut t')
+              = prim (PrimArray $ PrimArrayPutImmutable t')
```

Comment:
ok, fair enough. perhaps the arrays are only converted if they actually contain buffers